Sim/SimPapas/src/PapasAlg.cpp

```diff
@@ -1,34 +1,58 @@
 #include "PapasAlg.h"
-
+// FCCSW
+#include "SimPapas/IPapasExportTool.h"
+#include "SimPapas/IPapasImportTool.h"
 #include "SimPapas/IPapasTool.h"
+#include "SimPapasDetector/IPapasDetSvc.h"
+#include "papas/detectors/Detector.h"
+// PAPAS
+#include "papas/utility/Logger.h"
 #include "papas/utility/PDebug.h"
 #include "papas/utility/TRandom.h"
 
 #include <iostream>
+#define WITHSORT 1
 DECLARE_COMPONENT(PapasAlg)
 
 PapasAlg::PapasAlg(const std::string& name, ISvcLocator* svcLoc) : GaudiAlgorithm(name, svcLoc) {
   declareProperty("tools", m_toolNames);
+  declareProperty("importTool", m_importToolName);
+  declareProperty("exportTool", m_exportToolName);
   m_eventno = 0;
 }
 
 StatusCode PapasAlg::initialize() {
-
   for (auto& toolname : m_toolNames) {
     m_tools.push_back(tool<IPapasTool>(toolname));
   }
+  m_importTool = tool<IPapasImportTool>(m_importToolName);
+  m_exportTool = tool<IPapasExportTool>(m_exportToolName);
+  SmartIF<IPapasDetSvc> papasDetSvc;  // curiously does not work if joined with following line
```

Comment:
Could this be because one is forced to use an explicit constructor? 
I guess that 

    SmartIF<IPapasDetSvc> papasDetSvc(service(m_detServiceName))

should work? but no big deal anyway